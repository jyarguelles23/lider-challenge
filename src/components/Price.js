import React from "react";
import CincuentaPorciento from "../assets/50porciento.png";
import { formatAmount } from "../utils/MoneyFormat";

const Price = (priced) => {
  const { price, oldPrice } = priced;

  return (
    <div className="product-price">
      $ {formatAmount(price)}
      {oldPrice === 0 ? null : (
        <img src={CincuentaPorciento} alt="50%" className="image-porciento" />
      )}
    </div>
  );
};

export default Price;
