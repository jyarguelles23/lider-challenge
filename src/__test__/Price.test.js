import Price from "../components/Price";
import React from "react";
import renderer, { ReactTestRenderer } from "react-test-renderer";
let tree = ReactTestRenderer;

const price = {price : 400.000, oldPrice : 800.000};

describe("<Price/>", () => {
  beforeEach(() => {
    tree = renderer.create(<Price price={price} />);
  });

  it("should render without crashing", () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });
   
  it("should render an image of 50 %", () => {
    const precio = tree.root;
    
    expect(
      precio.findByProps({ className: "image-porciento" })
    ).toBeTruthy();
    
  });

});
