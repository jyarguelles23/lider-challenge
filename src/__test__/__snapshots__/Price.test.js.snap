// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Price/> should render without crashing 1`] = `
<div
  className="product-price"
>
  $ 
  [object Object]
  <img
    alt="50%"
    className="image-porciento"
    src="50porciento.png"
  />
</div>
`;
